casper/src/main/scala/io/casperlabs/casper/MultiParentCasperImpl.scala

```diff
@@ -365,7 +365,7 @@ class MultiParentCasperImpl[F[_]: Sync: Log: Metrics: Time: FinalityDetector: Bl
         for {
           dag            <- dag
           latestMessages <- dag.latestMessages
-          tipHashes      <- estimator(dag, latestMessages.mapValues(_.messageHash))
+          tipHashes      <- estimator(dag, latestMessages.mapValues(_.map(_.messageHash).toSet))
```

Comment:
```suggestion
          tipHashes      <- estimator(dag, latestMessages.mapValues(_.map(_.messageHash)))
```
At least it looks like it's a `Map[Validator, Set[Message]]` so the `.map` whould keep it as a `Set`.